src/js/components/Button/Button.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react';
+import classNames from 'classnames';
+import Icon from '../Icon/Icon';
+import { IconNameType } from '../Icon/icons';
+
+// Size is an attribute of the input html element so there was a conflict with our size props.
+// Hence the need to omit this attribute
+type ButtonAttributes = Omit<React.HTMLProps<HTMLButtonElement>, 'size'>;
+
+export interface Props {
+  children?: React.ReactNode;
+  className?: string;
+  icon?: IconNameType;
+  iconPosition?: 'left' | 'right';
+  iconTheme?: string;
+  isBlock?: boolean;
+  isDisabled?: boolean;
+  isOutline?: boolean;
+  onClick?: (arg0: React.MouseEvent) => void;
+  size?: 'lg' | 'md' | 'sm';
+  tag?: keyof JSX.IntrinsicElements;
+  theme?: string;
+  type?: 'button' | 'submit' | 'reset';
+}
+
+const Button: React.FC<ButtonAttributes & Props> = ({
```

Comment:
I think the easiest solution is to not be so strict about the props and remove the `ButtonAttributes`.

Reply:
Then we can't add the attributes related to the HTML element like name or autofocus Same with badges. If we want to use them as a button we can't use the attributes related to it. Then the props `tag` lose its interest

Reply:
For example we have a story using Campaign card with as `a` 
```<Col sm={6} md={4} lg={3} className="mb-2 mb-lg-0">
        <CampaignCard
          title="Social media toolkit for charities"
          state="Live"
          cover="https://images.unsplash.com/photo-1531206715517-5c0ba140b2b8?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60"
          tag="a"
          href="http://foo.com"
        />
      </Col>```

Typescript is not happy with the href attribute